core/services/ardupilot_manager/flight_controller_detector/Detector.py

```diff
@@ -45,19 +46,24 @@ def is_navigator_r5_connected() -> bool:
         return None
 
     @staticmethod
-    def detect_serial_platform(port: SysFS) -> Optional[Platform]:
-        if port.product in ["Pixhawk1", "PX4 FMU v2.x"]:
-            return Platform.Pixhawk1
-        if port.product in ["Pixhawk4", "PX4 FMU v5.x"]:
-            return Platform.Pixhawk4
-        if port.manufacturer in ["ArduPilot", "3D Robotics"] and (port.product and not "BL" in port.product):
-            return Platform.GenericSerial
+    def detect_serial_platform(port: SysFS, include_bootloaders: bool) -> Optional[Platform]:
+        if port.product is not None and "BL" in port.product and not include_bootloaders:
```

Comment:
```suggestion
        if "BL" in port.product and not include_bootloaders:
```
There is no reason to check for `None` if `"BL" in` will already do the job for it.